src/hooks/Recoil_Hooks.js

```diff
@@ -232,6 +233,8 @@ function useInterface(): RecoilInterface {
     function useRecoilState<T>(
       recoilState: RecoilState<T>,
     ): [T, SetterOrUpdater<T>] {
+      if (!isRecoilValue(recoilState))
+        throw new Error('No atom or selector provided');
```

Comment:
Perhaps these checks could be wrapped in a check for `__DEV__` to keep the bundle size down in production.

Reply:
@acutmore good idea!